apps/ensindexer/src/handlers/Resolver.ts

```diff
@@ -338,10 +402,47 @@ export const makeResolverHandlers = ({ pluginName }: { pluginName: PluginName })
         node: Node;
         name: Hex;
         resource: number;
-        record?: Hex;
       }>;
     }) {
-      // subgraph ignores
+      // subgraph ignores this event
+      if (pluginName === PluginName.Subgraph) return;
+
+      const { node, name, resource } = event.args;
+
+      // we only index TXT records (resource id 16)
+      if (resource !== 16) return;
+
+      // parse the record's name, which is the key of the DNS record
+      const [, recordName] = decodeDNSPacketBytes(hexToBytes(name));
+
+      // invariant: recordName is always available and parsed correctly
+      if (!recordName) throw new Error(`Invalid DNSPacket, cannot parse name '${name}'.`);
+
+      // relevant keys end with .ens
+      if (!recordName.endsWith(".ens")) return;
+
+      // trim the .ens off the end to match ENS record naming
+      const key = recordName.slice(0, -4);
+
+      const id = makeResolverId(event.log.address, node);
+      const resolver = await upsertResolver(context, {
+        id,
+        domainId: node,
+        address: event.log.address,
+      });
+
+      // remove relevant key
+      await context.db
+        .update(schema.resolver, { id })
+        .set({ texts: (resolver.texts ?? []).filter((text) => text !== key) });
+
+      // log ResolverEvent
+      await context.db.insert(schema.textChanged).values({
+        ...sharedEventValues(context.network.chainId, event),
+        resolverId: id,
+        key,
+        value: null,
+      });
```

Comment:
Making exceptions based on the plugin makes me think we could inject the new logic behind an interface from a particular plugin. Meaning, the `subgraph` plugin would implement a no-op for that interface, and both `basenames` and `lineanames` would implement the interface with the exact logic you proposed above.

Thoughts on that approach?